### PR TITLE
Perform migration for Swift 3 Preview 6

### PR DESCRIPTION
**Pull request by zwaldowski**

### CODE REVIEW

**Comment by alexsporn** [comment]
See fix for failing tests https://github.com/bignerdranch/Freddy/pull/193

File: Sources/JSONSerializing.swift [hunk 1]
```diff
@@ -20,21 +20,21 @@ extension JSON {
     private func toNSJSONSerializationObject() -> AnyObject {
```
Comment (pbartolome) [inline Sources/JSONSerializing.swift]: is there any reason why this is `AnyObject` and not `Any`?
The parameter `obj` in the method `JSONSerialization.data` uses `Any` and not `AnyObject`

```
class func data(withJSONObject obj: Any, options opt: JSONSerialization.WritingOptions = []) throws -> Data
```

**Comment by zwaldowski** [comment]
Rebased this branch and `swift3`. Will merge as soon as Travis comes back.

**Comment by daveanderson** [comment]
With Xcode 8 released is this in a position to be retested and potentially merged?

**Comment by daveanderson** [comment]
Thank you :)

**Pull request merged**